handler.lua

```diff
@@ -165,19 +189,29 @@ function CacheHandler:body_filter(conf)
   if not ctx then
     return
   end
-
-  local chunk = ngx.arg[1]
-  local eof = ngx.arg[2]
   
-  local res_body = ctx and ctx.res_body or ""
-  res_body = res_body .. (chunk or "")
-  ctx.res_body = res_body
+  local chunk, eof = ngx.arg[1], ngx.arg[2]
+  local rt_body_chunks = ngx.ctx.rt_body_chunks
+  local rt_body_chunk_number = ngx.ctx.rt_body_chunk_number
+
   if eof then
-    local content = json_decode(ctx.res_body)
-    local value = { content = content, headers = ctx.headers }
-    local value_json = json_encode(value)
-    ngx.timer.at(0, red_set, ctx.cache_key, value_json, conf)
+      local body = table.concat(rt_body_chunks)
+      ngx.arg[1] = body
+      local value = { content = body, headers = ctx.headers, status_code = ngx.status }
+      local value_json = json_encode(value)
+      local ok, err = ngx.timer.at(0, red_set, ctx.cache_key, value_json, conf)
+      if not ok then
+        ngx.log(ngx.ERR, "[response-cache] failed to create timer: ", err)
+      end
+  else
+      rt_body_chunks[rt_body_chunk_number] = chunk
+      rt_body_chunk_number = rt_body_chunk_number + 1
+      ngx.arg[1] = nil
+      ngx.ctx.rt_body_chunks = rt_body_chunks
+      ngx.ctx.rt_body_chunk_number = rt_body_chunk_number
   end
 end
 
+CacheHandler.PRIORITY = 10
```

Comment:
What issues were you running into with the previous priority?

Reply:
no issues as such but setting a lower priority so that it has access to the body chunk after any modifications to the body by upstream filters. See https://github.com/wshirey/kong-plugin-response-cache/pull/10#discussion_r276531609